install_microshift.py

```diff
@@ -0,0 +1,178 @@
+import subprocess
+import time
+import host
+from logger import logger
+import os
+
+def cleanup_microshift(h: host.Host):
+    files = ["rhocp-4.14.toml", "fast-datapath.toml", "kickstart.ks"]
+
+    for file in files:
+        if os.path.exists(f"./{file}"):
+            os.remove(f"./{file}")
+
+    cmd = f"podman rm -f minimal-microshift-server"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    old_compose = h.run("composer-cli compose status").out
+    old_compose_splitlines = old_compose.splitlines()
+
+    for line in old_compose_splitlines[1:]:
+        compose_id = line.split(" ")[0] 
+        if 'FAILED' or 'FINISHED' not in line:
+            cmd = f" composer-cli compose cancel {compose_id}"
+            output = h.run(cmd)
+            logger.info(output.out)
+    
+        cmd = f"composer-cli compose delete {compose_id}"
+        output = h.run(cmd)
+        logger.info(output.out)
+
+def generate_kickstart(): 
+    insertion_point = 'cat << EOF > /etc/crio/openshift-pull-secret'
```

Comment:
There is a much better way to code this without using search and replace.

Jinja templating or other templating methods is much cleaner.

Reply:
okay. Will use other templating methods